test/functional/lua/fs_spec.lua

```diff
@@ -308,26 +309,122 @@ describe('vim.fs', function()
       )
     end)
 
-    it('works with UNC paths', function()
-      eq('//foo', vim.fs.normalize('//foo')) -- UNC path
-      eq('//foo/bar', vim.fs.normalize('//foo//bar////')) -- UNC path
-      eq('/foo', vim.fs.normalize('///foo')) -- Not a UNC path
-      eq('/', vim.fs.normalize('//')) -- Not a UNC path
-      eq('/', vim.fs.normalize('///')) -- Not a UNC path
-      eq('/foo/bar', vim.fs.normalize('/foo//bar////')) -- Not a UNC path
-    end)
-
-    if is_os('win') then
+    if not is_os('win') then
+      it('preserves leading double slashes in POSIX paths', function()
+        eq('//foo', vim.fs.normalize('//foo'))
+        eq('//foo/bar', vim.fs.normalize('//foo//bar////'))
+        eq('/foo', vim.fs.normalize('///foo'))
+        eq('/', vim.fs.normalize('//'))
+        eq('/', vim.fs.normalize('///'))
+        eq('/foo/bar', vim.fs.normalize('/foo//bar////'))
+      end)
+      it('allows backslashes on unix-based os', function()
+        eq('/home/user/hello\\world', vim.fs.normalize('/home/user/hello\\world'))
+      end)
+    else
+      it('preserves / after drive letters', function()
+        eq('C:/', vim.fs.normalize([[C:\]]))
+      end)
```

Comment:
Doesn't this test the same thing as `Last slash is not truncated from root drive` test later on?

Reply:
Good catch, removing that test